app/src/main/cpp/skyline/common/languages.h

```diff
@@ -0,0 +1,107 @@
+         * @example #define LANG_ENTRY(lang, code, sysIndex, appIndex, map) func(lang, code, sysIndex, appIndex, map) <br> LANGUAGES <br> #undef LANG_ENTRY
+         */
+        #define LANGUAGES                                                       \
+            LANG_ENTRY(Japanese, ja, 0, 2, MAP)                                 \
+            LANG_ENTRY(AmericanEnglish, en-us, 1, 0, MAP)                       \
+            LANG_ENTRY(French, fr, 2, 3, MAP)                                   \
+            LANG_ENTRY(German, de, 3, 4, MAP)                                   \
+            LANG_ENTRY(Italian, it, 4, 7, MAP)                                  \
+            LANG_ENTRY(Spanish, es, 5, 6, MAP)                                  \
+            LANG_ENTRY(Chinese, zh-CN, 6, 14, DONT_MAP)                         \
+            LANG_ENTRY(Korean, ko, 7, 12, MAP)                                  \
+            LANG_ENTRY(Dutch, nl, 8, 8, MAP)                                    \
+            LANG_ENTRY(Portuguese, pt, 9, 10, MAP)                              \
+            LANG_ENTRY(Russian, ru, 10, 11, MAP)                                \
+            LANG_ENTRY(Taiwanese, zh-TW, 11, 13, DONT_MAP)                      \
+            LANG_ENTRY(BritishEnglish, en-GB, 12, 1, MAP)                       \
+            LANG_ENTRY(CanadianFrench, fr-CA, 13, 9, MAP)                       \
+            LANG_ENTRY(LatinAmericanSpanish, es-419, 14, 5, MAP)                \
+            LANG_ENTRY(SimplifiedChinese, zh-Hans, 15, 14, MAP)                 \
+            LANG_ENTRY(TraditionalChinese, zh-Hant, 16, 13, MAP)                \
+            LANG_ENTRY(BrazilianPortuguese, pt-BR, 17, 10, DONT_MAP)
+
+        /**
+         * @brief Enumeration of system languages
+         * @url https://switchbrew.org/wiki/Settings_services#Language
+         */
+        enum class SystemLanguage : u32 {
+            #define LANG_ENTRY(lang, code, sysIndex, appIndex, map) lang = sysIndex,
+            LANGUAGES
+            #undef LANG_ENTRY
+        };
+
+        #define LANG_ENTRY(lang, code, sysIndex, appIndex, map) ENUM_CASE(lang);
+        ENUM_STRING(SystemLanguage, LANGUAGES)
+        #undef LANG_ENTRY
+
+        /**
+         * @brief Enumeration of application languages
+         * @url https://switchbrew.org/wiki/NACP#ApplicationTitle
+         */
+        enum class ApplicationLanguage : u32 {
+            #define LANG_ENTRY(lang, code, sysIndex, appIndex, map) lang = appIndex,
+            LANGUAGES
+            #undef LANG_ENTRY
+        };
+
+        #define LANG_ENTRY(lang, code, sysIndex, appIndex, map) map(lang);
+        #define MAP(key) ENUM_CASE(key)
+        #define DONT_MAP(key)
+        ENUM_STRING(ApplicationLanguage, LANGUAGES)
+        #undef LANG_ENTRY
+        #undef MAP
+        #undef DONT_MAP
+
```

Comment:
using LanguageCode = u64

s/u64/LanguageCode

Reply:
I explicitly removed it [here](https://github.com/skyline-emu/skyline/pull/221/commits/3b02a06aa14a8abed8422927e567ece2dfb11aa3) as it's unneeded and only used in that file.

Reply:
I disagree, it makes it clearer to read

Reply:
added back in